### ISSUE TITLE
Binary Ninja plugin Manager can't find subdirectory plugin

### ISSUE THREAD

**Issue by ekilmer** [trailofbits/MUI#16]
We would like to set up MUI with a relatively normal Python package directory structure, but Binary Ninja plugin manager can't be configured to use a subdirectory in a Git repo.

We should open an issue to see how difficult it would be to do this.

Ethersplay also has this issue and directs users to install it manually.

**Comment by ekilmer** [comment]
Latest Binary Ninja dev build (`2.4.2917-dev`) now has support for plugins with subdirectories. Also a test branch [here](https://github.com/Vector35/community-plugins/commit/5f82b979a5c1df04aea596bb470db374b6e196d3#diff-4ce64887081bc2d8639d[…]0c32720f6bcdc20f491aaR3-R5) that I assume will work on this build now using the `subdirectory` json key (maybe wait until that branch is merged into the main branch of the repo?)

**Comment by psifertex** [comment]
Yup! This is now on stable and development builds as you can see from the Lighthouse plugin which exists in a subdirectory:

<img width="2032" alt="Screen Shot 2021-12-26 at 6 00 58 PM" src="https://user-images.githubusercontent.com/140215/147421982-0b6d9598-6cda-4a6c-bd8f-03c871e4abb1.png">

The short summary is to create a subfolder like: https://github.com/gaasedelen/lighthouse/tree/master/binjastub

Which contains the required `plugin.json`

Really, the only gotcha is making sure that import paths are adjusted appropriately otherwise it should "just work".

This would be great to make the plugin more discoverable to existing BN users.

**Comment by lordidiot** [comment]
Here is the workaround for testing the plugin manager locally https://github.com/Vector35/binaryninja-api/issues/2987#issuecomment-1054555686

**Comment by psifertex** [comment]
Sorry for not closing the loop earlier on this but it looks like the only thing that remains would be to have a [GH release](https://binary.ninja/2019/07/04/plugin-manager-2.0.html#5-create-a-release) that matches the version in the plugin.json. It also looks like (after an admittedly cursory glance) that [requirements.txt](https://github.com/trailofbits/ManticoreUI/blob/master/plugins/binaryninja/binjastub/requirements.txt) should list Z3 but doesn't and also I'm a bit concerned that the [import mui](https://github.com/trailofbits/ManticoreUI/blob/master/plugins/binaryninja/binjastub/__init__.py#L21) might fail due to load paths when in the plugin manager. That said, if ya'll have been testing it on a private plugin repo already maybe that's not an issue in which case, once you create the tag I can submit it in the main plugin manager.

Filed as a comment here since it seemed relevant but happy to make a new issue if ya'll would prefer.

**Comment by ekilmer** [comment]
@psifertex Thank you for following up! We are in the process of releasing a blog post for ManticoreUI. Once that is ready to publish, I will tag a release and let you know and/or make a PR to your plugin repo.

**Comment by psifertex** [comment]
Awesome. Happy to help if you've got any questions and looking forward to seeing the blog. Would be great to make it a one-click install for people who are interested.
